src/snowflake/snowpark/dataframe.py

```diff
@@ -433,25 +433,34 @@ def stat(self) -> DataFrameStatFunctions:
         return self._stat
 
     @df_action_telemetry
-    def collect(self) -> List["Row"]:
+    def collect(
+        self, *, _statement_params: Optional[Dict[str, Any]] = None
```

Comment:
How about removing the `_` since it's publicly available?

Reply:
I personally like the idea of removing `_`, but on the other side it breaks consistency with the connector library.

why we name those parameters prefix with `_` in the connector? is it for noting potential future breaking changes?

Reply:
@sfc-gh-mkeller may have more context

Reply:
I removed the `_`